builtin/providers/aws/resource_aws_instance.go

```diff
@@ -986,7 +997,15 @@ func buildAwsInstanceOpts(
 		}
 	}
 
-	if hasSubnet && associatePublicIPAddress {
+	if hasNetworkInterface {
+		ni := &ec2.InstanceNetworkInterfaceSpecification{
+			NetworkInterfaceId: aws.String(networkInterfaceID),
+			DeviceIndex:        aws.Int64(int64(0)),
+			SubnetId:           aws.String(subnetID),
```

Comment:
@catsby -
I'm not sure the best way to proceed on this.
When creating an ENI, you can specify `private_ip` and `security_groups`, however it seems that when assigning that to an Instance, if you also specify `security_groups` then the groups on the instance will replace the groups on the ENI.
However, setting the private_ip will not replace the existing private_ip on the ENI (but does not cause an error).
Setting subnet_id causes an error to occur.

I'm thinking the best way is that if specifying `network_interface_id`, then we disallow `subnet_id`, `security_group`, `vpc_security_group_id`, and `private_ip` from being specified and force their configuration via the ENI. Thoughts?